tools/cldr-code/src/main/java/org/unicode/cldr/util/SupplementalCalendarData.java

```diff
@@ -0,0 +1,269 @@
+        static final int INDEX = 4; // index of our element
+
+        /** the <era> xpath */
+        private final String start;
+
+        private final String end;
+        private final String code;
+        private final int type;
+        private final String calendarType; // for comparison
+        private final GregorianCalendar startCalendar;
+        private final GregorianCalendar endCalendar;
+
+        EraData(final XPathValue xpath) {
+            calendarType = xpath.getAttributeValue(2, LDMLConstants.TYPE);
+            type = Integer.parseInt(xpath.getAttributeValue(INDEX, LDMLConstants.TYPE));
+
+            start = xpath.getAttributeValue(INDEX, LDMLConstants.START);
+            startCalendar = forDateString(start);
+            end = xpath.getAttributeValue(INDEX, LDMLConstants.END);
+            endCalendar = forDateString(end);
+            code = xpath.getAttributeValue(INDEX, LDMLConstants.CODE);
+        }
+
+        public int getType() {
+            return type;
+        }
+
+        public String getStart() {
+            return start;
+        }
+
+        public String getEnd() {
+            return end;
+        }
+
+        public GregorianCalendar getStartCalendar() {
+            return startCalendar;
+        }
+
+        public GregorianCalendar getEndCalendar() {
+            return endCalendar;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public String[] getAliases() {
+            final String s = xpath.getAttributeValue(INDEX, LDMLConstants.ALIASES);
+            if (s == null) return "".split(" "); // empty array
+            return s.split(" ");
+        }
+
+        @Override
+        public String toString() {
+            return String.format(
+                    "ERA %d, [%s-%s] code=%s, aliases=%s",
+                    getType(), getStart(), getEnd(), getCode(), String.join(",", getAliases()));
+        }
+
+        private static final GregorianCalendar forDateString(String ymd) {
+            if (ymd == null) return null;
+            int multiplier = 1;
+            if (ymd.startsWith("-")) {
+                multiplier = -1;
+                ymd = ymd.substring(1);
+            }
+            final String[] parts = ymd.split("-");
+            try {
+                return new GregorianCalendar(
+                        multiplier * Integer.parseInt(parts[0]),
+                        Integer.parseInt(parts[1]) - 1,
+                        Integer.parseInt(parts[2]));
+            } catch (NumberFormatException nfe) {
+                throw new IllegalArgumentException("While parsing date string " + ymd, nfe);
+            }
+        }
+
+        private GregorianCalendar getLatest() {
+            GregorianCalendar l = getEndCalendar();
+            if (l == null) l = getStartCalendar();
+            return l;
+        }
+
+        /** only works within the same cal system */
+        @Override
```

Comment:
```suggestion
        @Override
        /* Normally comparisons require all of the fields to be listed, to ensure that no 
         * two unequal values a and b will never have a.compareTo(b) = 0.
         * However, the structure of the survey tool guaranties that no two eras (for different calendarTypes)
         * will overlap in start/end dates.
         */
```

Reply:
the survey tool doesn't guarantee this - do you mean the spec? the unit tests ought to validate this. But I'm not sure what all of the constraints are.

Reply:
Right, we don't make that clear. Filed https://unicode-org.atlassian.net/browse/CLDR-18576

Reply:
You're right. It is the spec not the Survey Tool.